pkg/cloud/externalconn/connection.go

```diff
@@ -22,14 +22,24 @@ import (
 // external resource.
 type Connection interface{}
 
+// ConnectionContext contains the dependencies passed to external connection
+// implementation during creation.
+type ConnectionContext interface {
+	// ExternalStorageContext returns the dependencies passed to external storage
+	// implementation during creation.
+	ExternalStorageContext() cloud.ExternalStorageContext
```

Comment:
This is a oneof situation, right? Is there any way of adding a hard-coding that? Or maybe just add a comment?

Reply:
Yeah it is expected that only one of these methods is implemented depending on who is `Dial`ing. I wonder if @miretskiy or you have any better ideas of designing this, I can definitely add a comment to that effect.

Reply:
I was initially thinking of just using types.Any -- but we wound up w/ oneOf...
Types.Any, of course would mean that the implementations of Dial would be responsible of ensuring that
the type they get is the expected type.